pkg/model/resource/resource.go

```diff
@@ -55,11 +57,18 @@ type Resource struct {
 
 // GVK returns the group-version-kind information to check against tracked resources in the configuration file
 func (r *Resource) GVK() config.GVK {
-	return config.GVK{
+	gvk := config.GVK{
 		Group:   r.Group,
 		Version: r.Version,
 		Kind:    r.Kind,
 	}
+
+	// Only store plural if it is irregular
+	if r.Plural != flect.Pluralize(strings.ToLower(r.Kind)) {
+		gvk.Plural = r.Plural
```

Comment:
WDYT about we add a scenario in the testdata?

Reply:
Once #1805 gets solved, we can add further changes. I just rebased the previously existing commit and removed the `v2` related part.